api/controllers/site.js

```diff
@@ -160,51 +160,52 @@ module.exports = {
     const siteId = Number(req.params.id);
 
     Promise.resolve(siteId).then((id) => {
-      if (isNaN(id)) {
+      if (_.isNaN(id)) {
         throw 404;
       }
       return Site.findByPk(id);
-    }).then((model) => {
-      site = model;
-      if (!site) {
-        throw 404;
-      }
-      return authorizer.update(req.user, site);
     })
-    .then(() => {
-      const params = Object.assign(site, req.body);
-      return site.update({
-        demoBranch: params.demoBranch,
-        demoDomain: params.demoDomain,
-        config: params.config,
-        previewConfig: params.previewConfig,
-        demoConfig: params.demoConfig,
-        defaultBranch: params.defaultBranch,
-        domain: params.domain,
-        engine: params.engine,
-      });
-    })
-    .then(model => Build.create({
-      user: req.user.id,
-      site: siteId,
-      branch: model.defaultBranch,
-    }))
-    .then(() => {
-      if (site.demoBranch) {
-        return Build.create({
-          user: req.user.id,
-          site: siteId,
-          branch: site.demoBranch,
+      .then((model) => {
+        site = model;
+        if (!site) {
+          throw 404;
+        }
+        return authorizer.update(req.user, site);
+      })
+      .then(() => {
+        const params = Object.assign(site, req.body);
+        return site.update({
+          demoBranch: params.demoBranch,
+          demoDomain: params.demoDomain,
+          config: params.config,
+          previewConfig: params.previewConfig,
+          demoConfig: params.demoConfig,
+          defaultBranch: params.defaultBranch,
+          domain: params.domain,
+          engine: params.engine,
         });
-      }
-      return null;
-    })
-    .then(() => siteSerializer.serialize(site))
-    .then((siteJSON) => {
-      res.json(siteJSON);
-    })
-    .catch((err) => {
-      res.error(err);
-    });
+      })
+      .then(model => Build.create({
+        user: req.user.id,
+        site: siteId,
+        branch: model.defaultBranch,
+      }))
+      .then(() => {
+        if (site.demoBranch) {
+          return Build.create({
+            user: req.user.id,
+            site: siteId,
+            branch: site.demoBranch,
+          });
+        }
+        return null;
```

Comment:
return null or Promise.resolve() ? or is that the same thing? for consistency, i think the codebase uses Promise.resolve()

Reply:
This is linting again... But we can update to `Promise.resolve`